crates/ra_ide/src/call_hierarchy.rs

```diff
@@ -0,0 +1,337 @@
+//! Entry point for call-hierarchy
+
+use indexmap::IndexMap;
+
+use hir::db::AstDatabase;
+use ra_syntax::{
+    ast::{self, DocCommentsOwner},
+    match_ast, AstNode, TextRange,
+};
+
+use crate::{
+    call_info::FnCallNode,
+    db::RootDatabase,
+    display::{ShortLabel, ToNav},
+    expand::descend_into_macros,
+    goto_definition, references, FilePosition, NavigationTarget, RangeInfo,
+};
+
+#[derive(Default)]
+struct CallLocations {
+    funcs: IndexMap<NavigationTarget, Vec<TextRange>>,
+}
+
+impl CallLocations {
+    pub fn add(&mut self, target: &NavigationTarget, range: TextRange) {
```

Comment:
These don't need to be `pub`. Also let's move this below `pub(crate)` function

Reply:
> Also let's move this below `pub(crate)` function

You mean move them below `CallItem`'s impl block?

Reply:
right, so that the public API of the module is on top